### ISSUE TITLE
Pushing not working: "LF will be replaced by CRLF"

### ISSUE THREAD

**Issue by uuuser-name** [denolehov/obsidian-git#37]
Whenever I try to push, this error shows up:
```
Uncaught (in promise) Error: warning: LF will be replaced by CRLF in .obsidian/plugins/obsidian-citation-plugin/main.js.
The file will have its original line endings in your working directory
warning: LF will be replaced by CRLF in .obsidian/plugins/obsidian-citation-plugin/styles.css.
The file will have its original line endings in your working directory
warning: LF will be replaced by CRLF in .obsidian/plugins/templater-obsidian/main.js.
The file will have its original line endings in your working directory
warning: LF will be replaced by CRLF in .obsidian/workspace.
The file will have its original line endings in your working directory
warning: LF will be replaced by CRLF in proton.md.
```
the error message goes on for lots of files, possibly every file in the vault, and ends on this:
```
warning: LF will be replaced by CRLF in fonction (math).mapp.js:1), <anonymous>:2304:85)
    at GitExecutorChain.eval (eval at <anonymous> (app.js:1), <anonymous>:2295:28)
    at Generator.throw (<anonymous>)
    at rejected (eval at <anonymous> (app.js:1), <anonymous>:2249:65)
```

The initial commit worked, and I checked and realized that no push ever worked since.
Any help/idea is appreciated.
on Windows 10 x64

**Comment by ebouchut** [comment]
Hi @uuuser-name 

I recently got the same error message from `git` on MacOS.
Here is my understanding of the problem and how I fixed this issue.

### Why?


TLDR;

Here is the long story:  http://adaptivepatchwork.com/2012/03/01/mind-the-end-of-your-line/

### Workaround

```
# Replace all `CR``LF` with `LF` (keeping dates unchanged)
find .obsidian/plugins -type f | xargs dos2unix -k

git add .obsidian
```
If you do not have `dos2unix` you can install with `brew` (a package manager), like so:
`brew install dos2unix`

### Solution

The ideal solution IMHO would be for plugin authors to use the following git configuration:

On **Windows**: `git config --global core.autocrlf true`
When committing a text file to the repository git replaces `CR`+`LF` with `LF`. 
When checking out a text file from the repository git replaces LF with CR+LF.
```
Working Directory =========> Repository
                              CRLF ==> LF 

Working Directory <========= Repository
                              CRLF <== LF 
```

On **Linux and MacOS**: `git config --global core.autocrlf input`
When committing a text file to the repository git replaces `CR`+`LF` with `LF`.
When checking out a text file from the repository git does not do any replacement.
```
Working Directory =========> Repository
                              CRLF ==> LF 

Working Directory <========= Repository
                              No Change 
```

### Sources

- https://stackoverflow.com/q/3206843/386517
- http://adaptivepatchwork.com/2012/03/01/mind-the-end-of-your-line/
- https://docs.github.com/en/github/using-git/configuring-git-to-handle-line-endings

**Comment by ebouchut** [comment]
As you are on **Windows**, you can give these `dos2unix` alternative a try:
- https://stackoverflow.com/questions/20368781/anything-like-dos2unix-for-windows
- https://koenaerts.ca/dos2unix-equivalent/

**Comment by Vinzent03** [comment]
Closing because of no response of @uuuser-name .

**Comment by jroachgit** [comment]
this is an installation setting when installing Git. What is the ideal setting when using Obsidian and GIT in Windows?

**Comment by ebouchut** [comment]
[This article from github](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings) gives IMHO a good overview of the suggested default for git users.
Make sure to select `Window` in the top menu beforehand.

**Comment by psarras** [comment]
Hey there. I am on windows as well and I am getting the same issue as above, although I can push no problem. ie. I have the setting @ebouchut  suggested set to true by default I constantly see files that I haven't changed appear as they are changed. I think obsidian is touching these files and messes with their original line endings. Whenever I commit those files, git is changing those back and discards the changes, therefore I am not committing anything on those files. As you can understand I am seeing ghost changes triggered from obsidian which is annoying. Could we have obsidian respecting the line endings instead of changing them to unix?

![image](https://user-images.githubusercontent.com/3820417/136004781-6f19aea8-c8fd-4ebd-ae8d-ae03551f5439.png)

**Comment by gitPrinz** [comment]
I came here for ghost changes and warnings.

My Problem was probably that I just used SmartGit and hadn't a Git Bash installed.
When installing this I think `git config --global core.autocrlf true` was set like suggested some times.
For now this seems to remove the ghost changes at least.

But sadly I still have a warning for every commit.

I am wondering why there is never a finale statement on any discussion clearly stating this:
E.G. in [the Forum](https://forum.obsidian.md/t/being-able-to-select-line-endings-crlf-lf/5294) there is also an unfinished discussion.
So it is probably not solved yet?

**Comment by gledos** [comment]
I'm about to be tortured crazy by this feature. It turns out that there is no simple solution to this problem, thank you.

**Comment by Neaxic** [comment]
This is seemingly a workaround, and im not sure it fits every case. However on a **windows** machine, try creating new folder, and new dummy files within that folder - if you can commit that, then make replica folders of those throwing issues, and move the files over.

That solved the issue for me, and i can use git obsidian now.

**Comment by alicenavorski** [comment]
I solved the problem that was caused with too long titles :
I tried  git config --global core.autocrlf true and it didn't change anything.
Then I tried to commit one file per one file through Source Control Pannel in Obsidian and the files that had an error were files with a title that was too long. I reduced the length, committed these files and tada ! Everything was pushed without a problem.
